lib/std/meta.zig

```diff
@@ -657,6 +657,8 @@ test "std.meta.tags" {
 pub fn FieldEnum(comptime T: type) type {
     const field_infos = fields(T);
 
+    if (field_infos.len == 0) return enum {};
```

Comment:
Stage1 doesn't allow empty enums so to pass CI you need to add a check for stage1 and use `@Type`.
```suggestion
    if (field_infos.len == 0) {
        if (@import("builtin").zig_backend == .stage1) @compileError("stage1 doesn't allow empty enums");
        return @Type(.{
            .Enum = .{
                .layout = .Auto,
                .tag_type = u0,
                .fields = &.{},
                .decls = &.{},
                .is_exhaustive = true,
            },
        });
    }
```

Adding a test for this case would also be nice, you can expand the already existing test on line 743.